swagger/swagger.js

```diff
@@ -48,75 +48,93 @@ module.exports = function(RED) {
+                            for(var i=0; i < swaggerPart.consumes.length; i++){
+                                swaggerPart.consumes[i] = swaggerPart.consumes[i].trim();
+                            }
                         }
-                    }
-                    if(swagger.deprecated){
-                        swaggerPart.deprecated = true;
-                    }
-                    if(swagger.parameters.length > 0){
-                        swaggerPart.parameters = swagger.parameters.slice();
-                        if(additionalParams){
-                            for(var i in additionalParams){
-                                swaggerPart.parameters.push(additionalParams[i]);
+                        if(swagger.produces){
+                            swaggerPart.produces = swagger.produces.split(',');
+                            for(var i=0; i < swaggerPart.produces.length; i++){
+                                swaggerPart.produces[i] = swaggerPart.produces[i].trim();
                             }
                         }
-                    } else if(additionalParams){
-                        swaggerPart.parameters = additionalParams.slice();
-                    }
-                    if(Object.keys(swagger.responses).length > 0){
-                        swaggerPart.responses = swagger.responses;
+                        if(swagger.deprecated){
+                            swaggerPart.deprecated = true;
+                        }
+                        if(swagger.parameters.length > 0){
+                            swaggerPart.parameters = swagger.parameters.slice();
+                            if(additionalParams){
+                                for(var i in additionalParams){
+                                    swaggerPart.parameters.push(additionalParams[i]);
+                                }
+                            }
+                        } else if(additionalParams){
+                            swaggerPart.parameters = additionalParams.slice();
+                        }
+                        if(Object.keys(swagger.responses).length > 0){
+                            swaggerPart.responses = swagger.responses;
+                        } else{
+                            swaggerPart.responses = {
+                                default: {}
+                            };
+                        }
+                        node.status({});
                     } else{
+                        swaggerPart.summary = node.name || (node.method+" "+url);
                         swaggerPart.responses = {
                             default: {}
                         };
+                        if(additionalParams){
+                            swaggerPart.parameters = additionalParams.slice();
+                        }
+                        node.status({fill:"yellow",shape:"ring",text:RED._("swagger.status.missingconfig")});
```

Comment:
The status text should just be the name of the message, `swagger.status.missingconfig` - don't translate it in the runtime side, let the editor do it for you. Otherwise the user gets the message in whatever language the runtime is using, not what language they are using in their browser.